share/ipld/nmt.go

```diff
@@ -65,212 +65,62 @@ func GetNode(ctx context.Context, bGetter blockservice.BlockGetter, root cid.Cid
 		return nil, err
 	}
 
-	return decodeBlock(block)
-}
-
-func decodeBlock(block blocks.Block) (ipld.Node, error) {
-	var (
-		leafPrefix  = []byte{nmt.LeafPrefix}
-		innerPrefix = []byte{nmt.NodePrefix}
-	)
-	data := block.RawData()
-	if len(data) == 0 {
-		return &nmtLeafNode{
-			cid:  cid.Undef,
-			Data: nil,
-		}, nil
-	}
-	domainSeparator := data[:typeSize]
-	if bytes.Equal(domainSeparator, leafPrefix) {
-		return &nmtLeafNode{
-			cid:  block.Cid(),
-			Data: data[typeSize:],
-		}, nil
-	}
-	if bytes.Equal(domainSeparator, innerPrefix) {
-		return &nmtNode{
-			cid: block.Cid(),
-			l:   data[typeSize : typeSize+nmtHashSize],
-			r:   data[typeSize+nmtHashSize:],
-		}, nil
-	}
-	return nil, fmt.Errorf(
-		"expected first byte of block to be either the leaf or inner node prefix: (%x, %x), got: %x)",
-		leafPrefix,
-		innerPrefix,
-		domainSeparator,
-	)
+	return nmtNode{Block: block}, nil
 }
 
 var _ ipld.Node = (*nmtNode)(nil)
```

Comment:
This is a good thing to do in general for non-ide users, but we are not doing this, so I think we can just remove this as well to clean up things completely.

Reply:
Removed compile time interface check.